File: go-tcc/src/app/graphql.go
package main

import (
	"bytes"
	"io"
	"log"
	"net/http"
	"time"
)

const GITHUB_GRAPHQL_URL = "https://api.github.com/graphql"

func doGraphQL(body string, token string) (int, int, time.Duration) {

	payload := bytes.NewBufferString(body)

	tamanho_request := payload.Len()

	req, _ := http.NewRequest(http.MethodPost, GITHUB_GRAPHQL_URL, payload)

	req.Header.Add("Content-Type", "application/json")
	req.Header.Add("Authorization", "Bearer "+token)

	time_init := time.Now()

	resp, err := client.Do(req)

	duration := time.Since(time_init)

	if err != nil {
		log.Printf("Erro ao executar requisição GraphQL: %v", err)
		return 0, 0, 0
	}

	defer resp.Body.Close()

	body_response, err := io.ReadAll(resp.Body)

	if err != nil {
		log.Printf("Erro lendo resposta GraphQL: %v", err)
		return 0, 0, 0
	}

	if resp.StatusCode != 200 {
		responseString := string(body_response)
		log.Printf("Erro ao executar requisição GraphQL, body: %v", responseString)
		return 0, 0, 0
	}

	return len(body_response), tamanho_request, duration
}

File: python-tcc/graphql.py
import json
import requests
import time

session = requests.Session()

GITHUB_GRAPHQL_URL = "https://api.github.com/graphql"

def do_graphql(body, token):

    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {token}"
    }

    payload_request = len(body.encode('utf-8'))

    try:
        time_init = time.time()
        response = session.post(GITHUB_GRAPHQL_URL, data=body, headers=headers, timeout=10)
        duration = time.time() - time_init
    except requests.exceptions.RequestException as e:
        print(f"Erro ao executar requisição GraphQL: {e}")
        return 0, 0, 0


    if response.status_code == 200:
        body_response = response.content
        return len(body_response), payload_request, duration * 1000
    else:
        print(f"Erro na resposta GraphQL: {response.status_code}")
        print(f"Body: {response.text}")
        return 0, 0, 0